commands/utils/quick-pick-azure.ts

```diff
@@ -40,19 +34,155 @@ export async function quickPickACRRepository(registry: Registry): Promise<Reposi
+}
+
+export async function quickPickLocation(subscription: Subscription): Promise<string> {
+    let locations: Location[] = await AzureUtilityManager.getInstance().getLocationsBySubscription(subscription);
+    let locationNames: string[] = [];
+
+    for (let loc of locations) {
+        locationNames.push(loc.displayName);
+    }
+
+    locationNames.sort((loc1: string, loc2: string): number => {
+        return loc1.localeCompare(loc2);
+    });
+
+    let location: string = await vscode.window.showQuickPick(locationNames, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a Location to use'
+    });
+    if (!location) { throw new UserCancelledError(); }
+    return location;
+}
+
+export async function quickPickResourceGroup(canCreateNew?: boolean, subscription?: Subscription): Promise<ResourceGroup> {
+    let resourceGroups = await AzureUtilityManager.getInstance().getResourceGroups(subscription);
+    let resourceGroupNames: string[] = [];
+
+    if (canCreateNew) { resourceGroupNames.push('+ Create new resource group'); }
+    for (let resGroupName of resourceGroups) {
+        resourceGroupNames.push(resGroupName.name);
+    }
+
+    let resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, {
+        'canPickMany': false,
+        'placeHolder': 'Choose a Resource Group to use'
+    });
+    if (!resourceGroupName) { throw new UserCancelledError(); }
+
+    let resourceGroup: ResourceGroup;
+    if (canCreateNew && resourceGroupName === '+ Create new resource group') {
+        if (!subscription) {
+            subscription = await quickPickSubscription();
+        }
+        const loc = await quickPickLocation(subscription);
+        resourceGroup = await createNewResourceGroup(loc, subscription);
+    } else {
+        resourceGroup = resourceGroups.find(resGroup => { return resGroup.name === resourceGroupName; });
+    }
+    return resourceGroup;
+}
+
+/** Requests confirmation for an action and returns true only in the case that the user types in yes
+ * @param yesOrNoPrompt Should be a yes or no question
+ */
+export async function confirmUserIntent(yesOrNoPrompt: string): Promise<boolean> {
+    //ensure user truly wants to delete image
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: true,
+        placeHolder: 'No',
+        value: 'No',
+        prompt: yesOrNoPrompt
+    };
+    let answer = await vscode.window.showInputBox(opt);
+    if (!answer) { throw new UserCancelledError(); }
+
+    answer = answer.toLowerCase();
+    return answer === 'yes';
+}
+
+/*Creates a new resource group within the current subscription */
+async function createNewResourceGroup(loc: string, subscription?: Subscription): Promise<ResourceGroup> {
+    const resourceGroupClient = AzureUtilityManager.getInstance().getResourceManagementClient(subscription);
+
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: false,
+        prompt: 'New resource group name?'
```

Comment:
capitalize Resource Group. The capitalization kinda goes back and forth between prompts. So why don't you stick with capitalizing the thing in question (ie Location, Registry, etc)

Reply:
Seems reasonable, I often do this unknowingly.